src/odemis/gui/cont/streams.py

```diff
@@ -1603,6 +1639,26 @@ def sem_capable():
             self.add_action("EBIC", self.addEBIC, sem_capable)
 
     def _onStreamUpdate(self, stream, updated):
+        if updated:
+            fv = self._tab_data_model.focussedView.value
+            if stream not in fv.stream_tree.flat.value:
+                # if the stream is hidden in the current focused view, then unhide
+                # it everywhere
+                for v in self._tab_data_model.views.value:
```

Comment:
Don't rely on the names of the views. Look at which Stream class is allowed to find out what to do. cf stream_classes and is_compatible()